src/Security/BeamNG.cpp

```diff
@@ -191,12 +234,76 @@ void LegitimacyCheck() {
             break;
         }
     }
+
+#elif defined(__APPLE__)
+    struct passwd* pw = getpwuid(getuid());
+    std::string homeDir = pw->pw_dir;
+    std::string crossoverBottlesPath = homeDir + "/Library/Application Support/CrossOver/Bottles/";
+    info("Crossover bottles path: " + crossoverBottlesPath);
+
+    for (const auto& bottle : std::filesystem::directory_iterator(crossoverBottlesPath)) {
+        if (!bottle.is_directory()) continue;
+
+        info("Checking bottle: " + bottle.path().filename().string());
+        auto driveMappings = GetDriveMappings(bottle.path().string());
+
+        std::string libraryFilePath = bottle.path().string() + "/drive_c/Program Files (x86)/Steam/config/libraryfolders.vdf";
+        std::ifstream libraryFile(libraryFilePath);
+        if (!libraryFile.is_open()) {
+            error("Failed to open libraryfolders.vdf in bottle '" + bottle.path().filename().string() + "'");
+            continue;
+        }
+        
+        auto root = tyti::vdf::read(libraryFile);
+        libraryFile.close();
+
+        for (const auto& [key, folderInfo] : root.childs) {
+            auto pathIter = folderInfo->attribs.find("path");
+            if (pathIter == folderInfo->attribs.end()) continue;
+
+            std::string path = pathIter->second;
+            info("Found Steam library path: " + path);
+
+            std::string driveLetter = Utils::ToLower(path.substr(0, path.find(":")));
+            driveLetter.erase(std::remove(driveLetter.begin(), driveLetter.end(), ':'), driveLetter.end());
+
+            if (driveMappings.find(driveLetter) == driveMappings.end()) {
+                warn("Drive letter " + driveLetter + " not found in mappings.");
+                continue;
+            }
+
+            std::string basePath = driveMappings[driveLetter];
+            if (!basePath.empty() && basePath.back() == '/') {
+                basePath.pop_back();
+            }
+
+            std::string additionalPath = path.substr(2);
+            std::replace(additionalPath.begin(), additionalPath.end(), '\\', '/');
+            if (!additionalPath.empty() && additionalPath.front() == '/') {
+                additionalPath.erase(0, 1);
+            }
+
+            std::filesystem::path beamngPath = std::filesystem::path(basePath) / additionalPath / "steamapps/common/BeamNG.drive";
+            info("Checking for BeamNG.drive in: " + beamngPath.string());
+
+            if (std::filesystem::exists(beamngPath)) {
+                info("BeamNG.drive found in bottle '" + bottle.path().filename().string() + "' at: " + beamngPath.string());
+                GameDir = beamngPath.string();
+                BottlePath = bottle.path().string();
+                info("GameDir: " + GameDir);
+                info("BottlePath: " + BottlePath);

```

Comment:
The game directory is already logged on line 290 here, please remove it. Also could the BottlePath log get a better message and could it be converted to a debug log instead.

Reply:
fix